sphinx/config.py

```diff
@@ -192,59 +181,18 @@ def __init__(self, dirname, filename, overrides, tags):
                 config['extensions'] = overrides.pop('extensions')
         self.extensions = config.get('extensions', [])  # type: List[unicode]
 
-        # correct values of copyright year that are not coherent with
-        # the SOURCE_DATE_EPOCH environment variable (if set)
-        # See https://reproducible-builds.org/specs/source-date-epoch/
-        if getenv('SOURCE_DATE_EPOCH') is not None:
-            for k in ('copyright', 'epub_copyright'):
-                if k in config:
-                    config[k] = copyright_year_re.sub(r'\g<1>%s' % format_date('%Y'),
-                                                      config[k])
+    @classmethod
+    def read(cls, confdir, filename, overrides=None, tags=None):
```

Comment:
I think we can go one step further and combine `confdir` and `filename` into something like a `path` attribute. There's no reason to keep these separate anymore, IMO.